tests/ui/search_is_some.rs

```diff
@@ -36,6 +36,11 @@ fn main() {
     // check that we don't lint if `find()` is called with
     // `Pattern` that is not a string
     let _ = "hello world".find(|c: char| c == 'o' || c == 'l').is_some();
+
+    // Check `find().is_some()`, single-line case.
+    let _ = (0..1).find(|x| **y == *x).is_some(); // one dereference less
+    let _ = (0..1).find(|x| *x == 0).is_some();
+    let _ = v.iter().find(|x| **x == 0).is_some();
```

Comment:
Could you add two more tests to see that the suggestion is correct:
1. One where the search value is used multiple times
    * Example: `(4..5).find(|x| *x == 1 || *x == 3 || *x == 5).is_some()`
2. One where a function takes `x` as is.
    * Example: `(1..3).find(|x| [1, 2, 3].contains(x)).is_some()`

And the same for `is_none()` at the end.

Reply:
Good tests, especially the second one. They will be added